scripts/zones/Crawlers_Nest/mobs/DE_Olavia.lua

```diff
@@ -0,0 +1,529 @@
+        { 261.0997, -32.4687, -28.7481 },
+        { 254.9180, -32.2382, -19.2297 },
+        { 246.3533, -32.3331, -20.5350 },
+        { 231.4488, -32.3491, -20.3175 },
+        { 215.1779, -32.4704, -18.9185 },
+        { 202.5445, -32.1740, -19.7428 },
+        { 193.4743, -32.2951, -18.3884 },
+        { 186.7378, -32.3773, -16.1250 },
+        { 180.5833, -32.4501, -17.4265 },
+        { 171.5948, -32.3908, -20.9317 },
+        { 155.0334, -32.1454, -18.3030 },
+        { 150.9698, -32.3579, -18.3564 },
+        { 140.2580, -32.3426, -23.4279 },
+        { 138.3673, -32.1804, -27.4211 },
+        { 139.9486, -32.1336, -33.7026 },
+        { 140.1400, -32.2996, -41.2877 },
+        { 141.0342, -34.2681, -48.2925 },
+        { 139.1214, -37.0170, -57.3631 },
+        { 132.8841, -39.7977, -66.5405 },
+        { 128.0140, -40.3259, -71.7650 }, -- First Drop
+    }
+}
+
+local path = paths.a -- Only one path works atm
+
+local escortProgress =
+{
+    NONE     = 0, -- Have not started pathing at all yet
+    ENROUTE  = 1, -- Started pathing and is continuing along path
+    PAUSED   = 2, -- Has paused pathing and awaiting orders
+    COMPLETE = 3, -- Has reached the end of the escort mission
+}
+
+local entity = {}
+
+entity.shouldMove = function(mob, progress)
+    return not mob:isFollowingPath() and mob:getStatus() == xi.status.NORMAL and progress ~= escortProgress.COMPLETE
+end
+
+entity.onMobInitialize = function(mob)
+    mob:setLocalVar('progress', escortProgress.NONE)
+    mob:setSpeed(60)
+
+    mob:setMobMod(xi.mobMod.NO_MOVE, 1)
+    mob:setMobMod(xi.mobMod.NO_DESPAWN, 1)
+    mob:setAutoAttackEnabled(true)
+end
+
+entity.onMobRoam = function(mob)
+    mob:setStatus(xi.status.NORMAL)
+    local progress = mob:getLocalVar('progress')
+    local escortID = '[Escort]Olavia'
+    local checkEscort = GetServerVariable(escortID)
+    local player = GetPlayerByID(mob:getLocalVar('player'))
+
+    local now = os.time()
+    local expire = mob:getLocalVar('expire')
+    if expire ~= 0 and expire <= now then
+        if progress ~= escortProgress.COMPLETE then
+            mob:showText(mob, ID.text.RAN_OUT_OF_TIME)
+        end
+
+        mob:setLocalVar('progress', escortProgress.NONE)
+        DespawnMob(mob:getID())
+        SetServerVariable(escortID, 0)
+        for _, v in ipairs(player:getParty()) do
+            utils.setQuestVar(v, 1, 70, 'Prog', 0)
+        end
+
+        return
+    else
+        local i = 0
+        if player ~= nil then
+            for _, v in ipairs(player:getParty()) do
+                if v:getZoneID() == xi.zone.CRAWLERS_NEST then
+                    if
+                        utils.getQuestVar(v, 1, 70, 'Prog') >= 1 or
+                        utils.getQuestVar(v, 1, 70, 'Prog') <= 4
+                    then
+                        i = i + 1
+                    end
+                end
+            end
+
+            if i == 0 then -- Players in party on quest no longer in zone reset the escort
```

Comment:
Should this block (and others in other NPCs) be inside the player ~= nil check? I feel like it won't work like intended like that

Reply:
hmm it does work like intended because the player still online but probably won't work if the player is no longer online. i guess in onMobInitialize i can run the above for loop and capture the party in a list and iterate over it checking if they are in the zone to handle disconnects.

Reply:
my original idea did not work but this seems to be a better solution
```
local party = {} 
entity.onMobRoam = functiion(mob)     
     local i = 0
        if next(party) ~= nil then
            for _, v in ipairs(party) do
                if v:getZoneID() == xi.zone.CRAWLERS_NEST then
                    i = i + 1
                end
            end

            if i == 0 then -- Players in party on quest no longer in zone reset the escort
                DespawnMob(checkEscort)
                SetServerVariable(escortID, 0)
                for _, v in ipairs(party) do
                    utils.setQuestVar(v, 1, 70, 'Prog', 0)
                end
            end
        end```

Reply:
Only real problem with this is that `local party = {}` isn't really "attached" to anything and may only work because only one escort NPC can exist. Horizon solved a similar problem by storing all players in a volatile server var. But I think for this case, they can be stored as localvars to the escort NPCs themselves.

These quests are such corner cases that I think this is ok.

Reply:
I had another thought -- What actually "attaches" these escort NPCs? the first player who zones in? Does it then update to include all eligible party members? hmm...

Reply:
Yea when the player zones in and get the cs it prevents anyone else zoning in and getting the cs. Only the person who gets the cs can start the escort. If your not in the zone when the escort starts your not eligible to complete it anymore and have to do it again or reset it by zoning out. 

For the storing of the party and such I don't really like the table solution personally but like you said only one can be up so it should be fine. I was trying avoid using to many local vars cause it's tough to keep track of them.

I was on the verge of making it its own system but I'm not sure the effort was worth it.